src/pages/patientView/mutationalSignatures/MutationalSignatureBarChart.tsx

```diff
@@ -0,0 +1,346 @@
+            return { ...obj, referenceTransformed };
+        });
+        return transformedDataSet;
+    });
+}
+
+@observer
+export default class MutationalBarChart extends React.Component<
+    IMutationalBarChartProps,
+    {}
+> {
+    constructor(props: IMutationalBarChartProps) {
+        super(props);
+    }
+
+    @computed get xTickLabels(): string[] {
+        return getColorsForSignatures(this.props.data).map(item => item.label);
+    }
+
+    @computed get yAxisDomain(): number[] {
+        const maxValue = this.props.data.reduce(
+            (previous: IMutationalCounts, current: IMutationalCounts) => {
+                return current.value > previous.value ? current : previous;
+            }
+        );
+        const minValue = this.props.data.reduce(
+            (previous: IMutationalCounts, current: IMutationalCounts) => {
+                return current.value < previous.value ? current : previous;
+            }
+        );
+        return [minValue.value, maxValue.value + 0.1 * maxValue.value];
+    }
+
+    @computed get formatLegendTopAxisPoints() {
+        const groupedData = _.groupBy(
+            getColorsForSignatures(this.props.data),
+            'group'
+        );
+        const labels = Object.keys(groupedData);
+        const labelObjects = getColorsForSignatures(this.props.data).map(
+            entry => ({
+                group: entry.group,
+                label: entry.label,
+                color: entry.colorValue,
+                subcategory: entry.subcategory,
+            })
+        );
+        const legendObjects = getLegendEntriesBarChart(labelObjects);
+        const lengthLegendObjects = getCenterPositionLabelEntries(
+            legendObjects
+        );
+        const legendOjbectsToAdd = createLegendLabelObjects(
+            lengthLegendObjects,
+            legendObjects,
+            labels
+        );
+        const centerOfBoxes = this.formatColorBoxLegend;
+        const xScale = getxScalePoint(labelObjects, 60, 1040);
+        const legendLabelsChart: JSX.Element[] = [];
+        legendOjbectsToAdd.forEach((item, i) => {
+            legendLabelsChart.push(
+                <VictoryLabel
+                    x={
+                        this.props.version != 'ID'
+                            ? centerOfBoxes[i].props.x +
+                              0.5 * centerOfBoxes[i].props.width
+                            : xScale(item.value)
+                    }
+                    y={8}
+                    width={this.props.width}
+                    text={item.group}
+                    style={{ fontSize: '13px', padding: 5 }}
+                    textAnchor={'middle'}
+                />
+            );
+        });
+        return legendLabelsChart;
+    }
+
+    @computed get formatColorBoxLegend() {
+        const groupedData = _.groupBy(
+            getColorsForSignatures(this.props.data),
+            'group'
+        );
+        const labels = Object.keys(groupedData);
```

Comment:
is there a better name for labels?  to distinguish from legendLabels

Reply:
Renamed `labels` to `mutationalSignaturesGroups`